[NEMO-269] Direct translation from Beam DAG to Nemo DAG

johnyangk opened the pull request:
JIRA: [NEMO-269: Direct translation from Beam DAG to Nemo DAG](https://issues.apache.org/jira/projects/NEMO/issues/NEMO-269)

**Major changes:**
- Direct translation from Beam DAG to Nemo DAG using CompositeBehavior
- Disables the previous partial combining optimization, to prepare for the new accumulator-based optimization

taegeonum commented on compiler/frontend/beam/src/main/java/org/apache/nemo/compiler/frontend/beam/PipelineTranslator.java:
isBatch -> isGlobalWindow? because batch can also use non-global windows

taegeonum commented on compiler/test/src/test/java/org/apache/nemo/compiler/frontend/beam/BeamFrontendALSTest.java:
```diff
@@ -36,27 +36,23 @@
 public final class BeamFrontendALSTest {
-  @Test
+  // TODO #260: Beam Accumulator-based Partial Aggregation
+  // @Test
```
?

taegeonum replied:
We don't have to disable it in this issue because this issue is just refactoring. No logic changed

taegeonum commented on compiler/test/src/test/java/org/apache/nemo/compiler/frontend/beam/BeamFrontendALSTest.java:
```diff
@@ -36,27 +36,23 @@
 
     assertEquals(producedDAG.getTopologicalSort(), producedDAG.getTopologicalSort());
-    assertEquals(42, producedDAG.getVertices().size());
+    assertEquals(38, producedDAG.getVertices().size());
```
Why is it changed?

taegeonum commented on compiler/test/src/test/java/org/apache/nemo/compiler/frontend/beam/BeamFrontendMLRTest.java:
```diff
@@ -36,26 +36,22 @@
 public class BeamFrontendMLRTest {
-  @Test
+  // TODO #260: Beam Accumulator-based Partial Aggregation
+  // @Test
```
?

taegeonum commented on compiler/test/src/test/java/org/apache/nemo/compiler/optimizer/pass/compiletime/composite/TransientResourceCompositePassTest.java:
```diff
@@ -49,46 +49,19 @@ public void setUp() throws Exception {
 
-  @Test
+  // TODO #260: Beam Accumulator-based Partial Aggregation
+  // @Test
```
?

taegeonum commented on compiler/test/src/test/java/org/apache/nemo/compiler/optimizer/pass/compiletime/reshaping/LoopExtractionPassTest.java:
```diff
@@ -44,10 +44,11 @@ public void setUp() throws Exception {
 
-  @Test
+  // TODO #260: Beam Accumulator-based Partial Aggregation
+  // @Test
```
?

johnyangk commented:
@taegeonum I've addressed your comment on the method name.
Regarding the tests, they are disabled because this PR disables the previous partial combining optimization and thus changes the number of vertices in the DAGs used in the tests.

Please see:
https://github.com/apache/incubator-nemo/pull/150/files#diff-5b36549ea7bd9b8064024cfd8db046a8R282
